File: 792-BinarySearch/792-BinarySearch.js
// Last updated: 6/9/2025, 1:56:19 PM
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let l = 0 , r= nums.length - 1;
    while(l<=r)
    {
        let mid =l+ Math.floor((r-l)/2);
        if(nums[mid]===target)
        {
            return mid;
        }
        else if(nums[mid] < target)
        {
            l = mid + 1;
        }
        else {
            r = mid-1;
        }
    }
    return -1;
};

File: 792-BinarySearch/792-BinarySearch.cpp
// Last updated: 6/9/2025, 1:56:24 PM
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n = nums.size();
        int low = 0, high = n-1;
        
        while(low<=high)
        {
            int mid = (low+high)/2 ; 
            if(nums[mid]==target) return mid;
            else if(target > nums[mid]) low = mid+1;
            else high=mid-1;
        }
        return -1;
    }
};